internal/service/alarm_notification_handler.go

```diff
@@ -0,0 +1,287 @@
+func NewAlarmNotificationHandler() *alarmNotificationHandlerBuilder {
+	return &alarmNotificationHandlerBuilder{}
+}
+
+// SetLogger sets the logger that the handler will use to write to the log. This is mandatory.
+func (b *alarmNotificationHandlerBuilder) SetLogger(
+	value *slog.Logger) *alarmNotificationHandlerBuilder {
+	b.logger = value
+	return b
+}
+
+// SetLoggingWrapper sets the wrapper that will be used to configure logging for the HTTP clients
+// used to connect to other servers, including the backend server. This is optional.
+func (b *alarmNotificationHandlerBuilder) SetLoggingWrapper(
+	value func(http.RoundTripper) http.RoundTripper) *alarmNotificationHandlerBuilder {
+	b.loggingWrapper = value
+	return b
+}
+
+// SetCloudID sets the identifier of the O-Cloud of this handler. This is mandatory.
+func (b *alarmNotificationHandlerBuilder) SetCloudID(
+	value string) *alarmNotificationHandlerBuilder {
+	b.cloudID = value
+	return b
+}
+
+// SetExtensions sets the fields that will be added to the extensions.
+func (b *alarmNotificationHandlerBuilder) SetExtensions(
+	values ...string) *alarmNotificationHandlerBuilder {
+	b.extensions = values
+	return b
+}
+
+// SetExtensions sets the fields that will be added to the extensions.
+func (b *alarmNotificationHandlerBuilder) SetKubeClient(
+	kubeClient *k8s.Client) *alarmNotificationHandlerBuilder {
+	b.kubeClient = kubeClient
+	return b
+}
+
+// Build uses the data stored in the builder to create anad configure a new handler.
+func (b *alarmNotificationHandlerBuilder) Build(ctx context.Context) (
+	result *alarmNotificationHandler, err error) {
+	// Check parameters:
+	if b.logger == nil {
+		err = errors.New("logger is mandatory")
+		return
+	}
+	if b.cloudID == "" {
+		err = errors.New("cloud identifier is mandatory")
+		return
+	}
+
+	if b.kubeClient == nil {
+		err = errors.New("kubeClient is mandatory")
+		return
+	}
+
+	// Prepare the JSON iterator API:
+	jsonConfig := jsoniter.Config{
+		IndentionStep: 2,
+	}
+	jsonAPI := jsonConfig.Froze()
+
+	// Create the filter expression evaluator:
+	pathEvaluator, err := search.NewPathEvaluator().
+		SetLogger(b.logger).
+		Build()
+	if err != nil {
+		return
+	}
+	selectorEvaluator, err := search.NewSelectorEvaluator().
+		SetLogger(b.logger).
+		SetPathEvaluator(pathEvaluator.Evaluate).
+		Build()
+	if err != nil {
+		return
+	}
+	// Create the jq tool:
+	jqTool, err := jq.NewTool().
+		SetLogger(b.logger).
+		Build()
+	if err != nil {
+		return
+	}
+
+	// Check that extensions are at least syntactically valid:
+	for _, extension := range b.extensions {
+		_, err = jqTool.Compile(extension)
+		if err != nil {
+			return
+		}
+	}
+
+	alarmSubscriptionSearcher := newAlarmSubscriptionSearcher()
+	alarmSubscriptionSearcher.SetLogger(b.logger).SetJqTool(jqTool).build()
+
+	// create persist storeage option
+	persistStore := persiststorage.NewKubeConfigMapStore().
+		SetNameSpace(TestNamespace).
+		SetName(TestConfigmapName).
```

Comment:
TestConfigmapName is a constant just for current testings, right? It probably should be customizable later on.

Reply:
It is temperate for testConfigurationName. It will have a official name. Let me update this and propose a potential official name. Please see next commit/update.